contracts/Core/AssetManager.sol

```diff
@@ -82,37 +81,28 @@ contract AssetManager is AssetStorage, StateManager, AssetManagerParams, IAssetM
         uint32 epoch = _getEpoch(epochLength);
         if (assetStatus) {
             if (!collections[id].active) {
-                activeCollections.push(id);
-                collections[id].assetIndex = uint16(activeCollections.length);
+                // slither-disable-next-line incorrect-equality
+                if (updateRegistry == epoch) {
+                    // slither-disable-next-line reentrancy-events,reentrancy-no-eth,reentrancy-benign
```

Comment:
I have disabled the reentrancy slither issues on updateRegistry functions for now but we will need to address this. 
If we shift the registry from the delegator to the asset Manager, the reentrancy issues can be fixed.

@hrishikeshio @abhishekvispute @SkandaBhat